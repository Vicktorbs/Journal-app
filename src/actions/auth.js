import { firebase, googleAuthProvider } from "../firebase/firebase_config";
import { types } from "../types/types"

export const startLoginWithEmailPassword = (email, password) => {
    return (dispatch) => {
        setTimeout(() => {
            dispatch(login(12345, 'Juan'))
        }, 3500);
    }
}

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider).then(({ user }) => {
            dispatch(login(user.uid, user.displayName))
        })
    }
}

export const login = (uid, displayName) => {
    return {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    }
}