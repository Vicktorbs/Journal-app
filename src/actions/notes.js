import Swal from "sweetalert2";
import { db } from "../firebase/firebase_config";
import { loadNotes } from "../helpers/loadNotes";
import { types } from "../types/types";

export const startNewNote = () => {
    return async(dispatch, getState) => {
        const uid = getState().auth.uid;

        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        }

        const documentRef = await db.collection(`${ uid }/journal/notes`).add(newNote);
        // console.log(documentRef);
        dispatch(activeNote(documentRef.id, newNote));
        // funcionalidad personalizada, checar
        dispatch(startLoadingNotes(uid));
    }
}

export const activeNote = (id, note) => ({
    type: types.notesActive,
    payload: {
        id,
        ...note
    }
})

export const startLoadingNotes = (uid) => {
    return async(dispatch) => {
        const notes = await loadNotes(uid);
        dispatch(setNotes(notes))
    }
}

export const setNotes = (notes) => ({
    type: types.notesLoad,
    payload: notes
})

export const startSaveNote = (note) => {
    return async(dispatch, getState) => {
        const uid = getState().auth.uid;

        if (!note.url) {
            delete note.url
        }

        const noteToFirestore = { ...note };
        delete noteToFirestore.id;

        await db.doc(`${ uid }/journal/notes/${ note.id }`).update(noteToFirestore);

        dispatch(refresNote(note.id, noteToFirestore));
        Swal.fire('Save', note.title, 'success')
    }
}

export const refresNote = (id, note) => ({
    type: types.notesUpdated,
    payload: {
        id,
        note: {
            id,
            ...note
        }
    }
})