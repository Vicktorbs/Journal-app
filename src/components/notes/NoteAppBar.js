import React from 'react'

export const NoteAppBar = () => {
    return (
        <div className="notes_appbar">
            <span>25 de Abril 2021</span>

            <div>
                <button className="btn">
                    Picture
                </button>
                <button className="btn">
                    Save
                </button>
            </div>
        </div>
    )
}
