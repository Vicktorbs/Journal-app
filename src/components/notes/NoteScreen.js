import React from 'react'
import { NoteAppBar } from './NoteAppBar'

export const NoteScreen = () => {
    return (
        <div className="notes__main-content">
            <NoteAppBar />

            <div className="notes__content">
                <input type="text" placeholder="Some awesome title" className="notes__title-input" />

                <textarea placeholder="Whant happened today?" className="notes_text-area"></textarea>

                <div className="notes__images">
                    <img alt="Upload" src="https://i.pinimg.com/originals/c3/72/2c/c3722cfbfbe16b617bc55c547bb72691.jpg" />
                </div>
            </div>

        </div>
    )
}
