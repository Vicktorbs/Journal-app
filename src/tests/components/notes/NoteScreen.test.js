import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; 
import thunk from 'redux-thunk';
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';


jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        uid: '464',
        name: 'victor'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 1234,
            title: 'hola',
            body: 'mundo',
            date: 0
        },
        notes: []
    }
}

let store = mockStore(initialState);
store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={ store }>
        <NoteScreen />
    </Provider>
)

describe('Test on comoponent <NoteScreen />', () => {
    
    test('should show itselft correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should lauch activeNote', () => {
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hello again'
            }
        })

        expect(activeNote).toHaveBeenCalled();
        expect(activeNote).toHaveBeenCalledWith(
            1234,
            {
                body: 'mundo',
                title: 'Hello again',
                id: 1234,
                date: 0
            }
        );
    })
    
})
