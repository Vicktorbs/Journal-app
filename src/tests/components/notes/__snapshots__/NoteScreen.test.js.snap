// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on comoponent <NoteScreen /> should show itselft correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes_appbar"
  >
    <span>
      25 de Abril 2021
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="hola"
    />
    <textarea
      className="notes_text-area"
      name="body"
      onChange={[Function]}
      placeholder="Whant happened today?"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
